device/cuda/include/cuda/seeding/seed_finding.hpp

```diff
@@ -10,15 +10,16 @@
 #include <algorithm>
 #include <cuda/seeding/detail/doublet_counter.hpp>
 #include <cuda/seeding/detail/multiplet_estimator.hpp>
-#include <cuda/seeding/doublet_counting.cuh>
-#include <cuda/seeding/doublet_finding.cuh>
-#include <cuda/seeding/seed_selecting.cuh>
-#include <cuda/seeding/triplet_counting.cuh>
-#include <cuda/seeding/triplet_finding.cuh>
-#include <cuda/seeding/weight_updating.cuh>
+#include <cuda/seeding/doublet_counting.hpp>
+#include <cuda/seeding/doublet_finding.hpp>
+#include <cuda/seeding/seed_selecting.hpp>
+#include <cuda/seeding/triplet_counting.hpp>
+#include <cuda/seeding/triplet_finding.hpp>
+#include <cuda/seeding/weight_updating.hpp>
```

Comment:
I thought so. But they don't have any cuda related code inside .cuh so decided to use .hpp extension for them.

Reply:
This was one of my recommendations to use `.hpp` on headers that can be included by not only `.cu` files, but other C\+\+ source files as well.

Since as I explained before, I like to use `.cuh` as a "contagious" extension. Marking something as `.cuh` should be done only when that particular header has code for launching a kernel, or when it includes some other `.cuh` file. In all other cases we should just use `.hpp`.